### PR TITLE
Feature/i 30  send events to web

### PR DESCRIPTION
**Pull request by NPseba**

### CODE REVIEW

**Comment by NPseba** [comment]
It has unpredictable behavior and I can't figure out why. Sometimes the message reaches the browser and it is displayed accordingly but most of the time it does not

**Comment by NPseba** [comment]
Here are logs from console

`Navigated to http://127.0.0.1:5000/
VM923 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM933 application.js:12 Connection established
VM933 application.js:12 Connection established
VM929 socket.io.min.js:2 WebSocket connection to 'ws://127.0.0.1:5000/socket.io/?EIO=3&transport=websocket&sid=9bd3f07f0d9a40078e2a70852d5b9a86' failed: Received a broken close frame containing invalid UTF-8.
n @ VM929 socket.io.min.js:2
n.doOpen @ VM929 socket.io.min.js:2
n.open @ VM929 socket.io.min.js:1
r.probe @ VM929 socket.io.min.js:1
r.onOpen @ VM929 socket.io.min.js:1
r.onHandshake @ VM929 socket.io.min.js:1
r.onPacket @ VM929 socket.io.min.js:1
(anonymous) @ VM929 socket.io.min.js:1
n.emit @ VM929 socket.io.min.js:1
n.onPacket @ VM929 socket.io.min.js:1
n @ VM929 socket.io.min.js:2
(anonymous) @ VM929 socket.io.min.js:2
n.decodePayloadAsBinary @ VM929 socket.io.min.js:2
n.decodePayload @ VM929 socket.io.min.js:2
n.onData @ VM929 socket.io.min.js:2
(anonymous) @ VM929 socket.io.min.js:1
n.emit @ VM929 socket.io.min.js:1
i.onData @ VM929 socket.io.min.js:1
i.onLoad @ VM929 socket.io.min.js:1
hasXDR.e.onreadystatechange @ VM929 socket.io.min.js:1
VM933 application.js:12 Connection established
VM933 application.js:12 Connection established
VM933 application.js:12 Connection established
VM933 application.js:12 Connection established
VM933 application.js:12 Connection established
VM933 application.js:12 Connection established
VM933 application.js:12 Connection established
VM933 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM943 application.js:12 Connection established
VM943 application.js:12 detected traffic light
VM943 application.js:12 moving forward
Navigated to http://127.0.0.1:5000/
VM1074 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM1084 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM1094 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM1104 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM1114 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM1124 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM1134 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM1144 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM1176 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM1186 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
VM1196 application.js:12 Connection established
VM1192 socket.io.min.js:2 WebSocket connection to 'ws://127.0.0.1:5000/socket.io/?EIO=3&transport=websocket&sid=17da0bf6335c4578af534c0595d93175' failed: Received a broken close frame containing invalid UTF-8.
n @ VM1192 socket.io.min.js:2
n.doOpen @ VM1192 socket.io.min.js:2
n.open @ VM1192 socket.io.min.js:1
r.probe @ VM1192 socket.io.min.js:1
r.onOpen @ VM1192 socket.io.min.js:1
r.onHandshake @ VM1192 socket.io.min.js:1
r.onPacket @ VM1192 socket.io.min.js:1
(anonymous) @ VM1192 socket.io.min.js:1
n.emit @ VM1192 socket.io.min.js:1
n.onPacket @ VM1192 socket.io.min.js:1
n @ VM1192 socket.io.min.js:2
(anonymous) @ VM1192 socket.io.min.js:2
n.decodePayloadAsBinary @ VM1192 socket.io.min.js:2
n.decodePayload @ VM1192 socket.io.min.js:2
n.onData @ VM1192 socket.io.min.js:2
(anonymous) @ VM1192 socket.io.min.js:1
n.emit @ VM1192 socket.io.min.js:1
i.onData @ VM1192 socket.io.min.js:1
i.onLoad @ VM1192 socket.io.min.js:1
hasXDR.e.onreadystatechange @ VM1192 socket.io.min.js:1
VM1196 application.js:12 Connection established
Navigated to http://127.0.0.1:5000/
application.js:12 Connection established
application.js:12 Connection established
socket.io.min.js:2 WebSocket connection to 'ws://127.0.0.1:5000/socket.io/?EIO=3&transport=websocket&sid=cd62a582e78544238a3201272bca4fdd' failed: Received a broken close frame containing invalid UTF-8.
n @ socket.io.min.js:2
n.doOpen @ socket.io.min.js:2
n.open @ socket.io.min.js:1
r.probe @ socket.io.min.js:1
r.onOpen @ socket.io.min.js:1
r.onHandshake @ socket.io.min.js:1
r.onPacket @ socket.io.min.js:1
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
n.onPacket @ socket.io.min.js:1
n @ socket.io.min.js:2
(anonymous) @ socket.io.min.js:2
n.decodePayloadAsBinary @ socket.io.min.js:2
n.decodePayload @ socket.io.min.js:2
n.onData @ socket.io.min.js:2
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
i.onData @ socket.io.min.js:1
i.onLoad @ socket.io.min.js:1
hasXDR.e.onreadystatechange @ socket.io.min.js:1
application.js:12 Connection established
socket.io.min.js:2 WebSocket connection to 'ws://127.0.0.1:5000/socket.io/?EIO=3&transport=websocket&sid=430506613ab64e78b9ff901fbafca2ca' failed: Received a broken close frame containing invalid UTF-8.
n @ socket.io.min.js:2
n.doOpen @ socket.io.min.js:2
n.open @ socket.io.min.js:1
r.probe @ socket.io.min.js:1
r.onOpen @ socket.io.min.js:1
r.onHandshake @ socket.io.min.js:1
r.onPacket @ socket.io.min.js:1
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
n.onPacket @ socket.io.min.js:1
n @ socket.io.min.js:2
(anonymous) @ socket.io.min.js:2
n.decodePayloadAsBinary @ socket.io.min.js:2
n.decodePayload @ socket.io.min.js:2
n.onData @ socket.io.min.js:2
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
i.onData @ socket.io.min.js:1
i.onLoad @ socket.io.min.js:1
hasXDR.e.onreadystatechange @ socket.io.min.js:1
application.js:12 Connection established
application.js:12 uploaded image
socket.io.min.js:2 WebSocket connection to 'ws://127.0.0.1:5000/socket.io/?EIO=3&transport=websocket&sid=b0856120e53546a5b74d78a894a95efe' failed: Received a broken close frame containing invalid UTF-8.
n @ socket.io.min.js:2
n.doOpen @ socket.io.min.js:2
n.open @ socket.io.min.js:1
r.probe @ socket.io.min.js:1
r.onOpen @ socket.io.min.js:1
r.onHandshake @ socket.io.min.js:1
r.onPacket @ socket.io.min.js:1
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
n.onPacket @ socket.io.min.js:1
n @ socket.io.min.js:2
(anonymous) @ socket.io.min.js:2
n.decodePayloadAsBinary @ socket.io.min.js:2
n.decodePayload @ socket.io.min.js:2
n.onData @ socket.io.min.js:2
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
i.onData @ socket.io.min.js:1
i.onLoad @ socket.io.min.js:1
hasXDR.e.onreadystatechange @ socket.io.min.js:1
application.js:12 Connection established
socket.io.min.js:2 WebSocket connection to 'ws://127.0.0.1:5000/socket.io/?EIO=3&transport=websocket&sid=f8e6ad532e2a448ba8089f91b6cb3508' failed: Received a broken close frame containing invalid UTF-8.
n @ socket.io.min.js:2
n.doOpen @ socket.io.min.js:2
n.open @ socket.io.min.js:1
r.probe @ socket.io.min.js:1
r.onOpen @ socket.io.min.js:1
r.onHandshake @ socket.io.min.js:1
r.onPacket @ socket.io.min.js:1
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
n.onPacket @ socket.io.min.js:1
n @ socket.io.min.js:2
(anonymous) @ socket.io.min.js:2
n.decodePayloadAsBinary @ socket.io.min.js:2
n.decodePayload @ socket.io.min.js:2
n.onData @ socket.io.min.js:2
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
i.onData @ socket.io.min.js:1
i.onLoad @ socket.io.min.js:1
hasXDR.e.onreadystatechange @ socket.io.min.js:1
application.js:12 Connection established
socket.io.min.js:2 WebSocket connection to 'ws://127.0.0.1:5000/socket.io/?EIO=3&transport=websocket&sid=401c837b451643c986ed255868b5677f' failed: Received a broken close frame containing invalid UTF-8.
n @ socket.io.min.js:2
n.doOpen @ socket.io.min.js:2
n.open @ socket.io.min.js:1
r.probe @ socket.io.min.js:1
r.onOpen @ socket.io.min.js:1
r.onHandshake @ socket.io.min.js:1
r.onPacket @ socket.io.min.js:1
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
n.onPacket @ socket.io.min.js:1
n @ socket.io.min.js:2
(anonymous) @ socket.io.min.js:2
n.decodePayloadAsBinary @ socket.io.min.js:2
n.decodePayload @ socket.io.min.js:2
n.onData @ socket.io.min.js:2
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
i.onData @ socket.io.min.js:1
i.onLoad @ socket.io.min.js:1
hasXDR.e.onreadystatechange @ socket.io.min.js:1
application.js:12 Connection established
socket.io.min.js:1 GET http://127.0.0.1:5000/socket.io/?EIO=3&transport=polling&t=1590442758068-21 net::ERR_CONNECTION_REFUSED
i.create @ socket.io.min.js:1
i @ socket.io.min.js:1
o.request @ socket.io.min.js:1
o.doPoll @ socket.io.min.js:1
n.poll @ socket.io.min.js:2
n.doOpen @ socket.io.min.js:2
n.open @ socket.io.min.js:1
r.open @ socket.io.min.js:1
r @ socket.io.min.js:1
r @ socket.io.min.js:1
n.open.n.connect @ socket.io.min.js:1
(anonymous) @ socket.io.min.js:1
setTimeout (async)
n.reconnect @ socket.io.min.js:1
n.onclose @ socket.io.min.js:1
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
r.onClose @ socket.io.min.js:1
(anonymous) @ socket.io.min.js:1
n.emit @ socket.io.min.js:1
n.onClose @ socket.io.min.js:1
n.onClose @ socket.io.min.js:2
ws.onclose @ socket.io.min.js:2
Navigated to chrome-error://chromewebdata/
Navigated to http://127.0.0.1:5000/
application.js:12 Connection established
`

File: rpi/boundary/car_event_processor.py [hunk 1]
```diff
@@ -23,29 +23,40 @@ def __init__(self):
 
         dispatcher.connect(self.move_forward, signal=CarStatus.MOVE_FORWARD, sender=dispatcher.Any)
 
+        self._socketio = socketio
+
     def take_picture(self):
```
Comment (dd-iuonac) [inline rpi/boundary/car_event_processor.py:28]: Try to use **emit** instead of **send**.
For me **send** doesn't work.

Comment (NPseba) [reply rpi/boundary/car_event_processor.py:28]: Unfortunately, it doesn't make any difference. I also did try to set the broadcast parameter to true or to emit on a specific event and use that event on client side. It simply just appear sometimes in browser. I did try to look in app console, after activating the logs, and browser console to see if I see anything special occurs but I did not see

**Pull request merged**